corehq/apps/consumer_user/forms/consumer_user_authentication_form.py

```diff
@@ -0,0 +1,48 @@
+from django import forms
+from django.contrib.auth.forms import AuthenticationForm
+from django.core.exceptions import ValidationError
+from django.utils.translation import ugettext_lazy as _
+
+from corehq.apps.consumer_user.const import (
+    CONSUMER_INVITATION_ACCEPTED,
+    CONSUMER_INVITATION_STATUS,
+)
+from corehq.apps.consumer_user.models import (
+    ConsumerUser,
+    ConsumerUserCaseRelationship,
+)
+from corehq.apps.domain.forms import NoAutocompleteMixin
+from corehq.apps.hqcase.utils import update_case
+
+
+class ConsumerUserAuthenticationForm(NoAutocompleteMixin, AuthenticationForm):
+    username = forms.EmailField(label=_("Email Address"),
+                                widget=forms.TextInput(attrs={'class': 'form-control'}),
+                                required=True)
+    password = forms.CharField(label=_("Password"),
+                               widget=forms.PasswordInput(attrs={'class': 'form-control'}),
+                               required=True)
+
+    def __init__(self, *args, **kwargs):
+        self.invitation = kwargs.pop('invitation', None)
+        super().__init__(*args, **kwargs)
+
+    def clean(self):
+        username = self.cleaned_data.get('username')
+        if username is None:
+            raise ValidationError(_('Please enter a valid email address.'))
+
+        password = self.cleaned_data.get('password')
+        if not password:
+            raise ValidationError(_("Please enter a password."))
+        cleaned_data = super(ConsumerUserAuthenticationForm, self).clean()
```

Comment:
The rest of this method should probably happen in `.save()` instead, since it's making changes, not validating or cleaning data.

Reply:
Actually, looking at this closer, why do those modifications happen here?  This is an authentication form, but it might also accept an invitation and create a new consumer user.  Shouldn't that be handled separately?